### PR TITLE
Opaque pointers: Update tests - part 1

### PR DESCRIPTION
**Pull request by mariusz-sikora-at-amd**

### CODE REVIEW

**Comment by mariusz-sikora-at-amd** [comment]
Before updating all needed llvm-lit tests I would like to get your feedback if this is a good approach. For transition time we will be able to tests both versions, with and without opaque pointers.
Later the non opaque pointer version will be removed.

**Comment by amdvlk-admin** [comment]
### Test summary for commit f589cbef88e2e64cd3355a2aca0ded21fe751256
<details><summary>CTS tests (Failed: 0/286534)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.6, Gfx8</summary><ul>

<li>Passed: 36222/72653 (49.9%)</li>
<li>Failed: 0/72653 (0.0%)</li>
<li>Not Supported: 36431/72653 (50.1%)</li><li>Warnings: 0/72653 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 40420/72700 (55.6%)</li>
<li>Failed: 0/72700 (0.0%)</li>
<li>Not Supported: 32280/72700 (44.4%)</li><li>Warnings: 0/72700 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 37135/72707 (51.1%)</li>
<li>Failed: 0/72707 (0.0%)</li>
<li>Not Supported: 35572/72707 (48.9%)</li><li>Warnings: 0/72707 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 34834/68474 (50.9%)</li>
<li>Failed: 0/68474 (0.0%)</li>
<li>Not Supported: 33640/68474 (49.1%)</li><li>Warnings: 0/68474 (0.0%)</li>
</ul></details>


</ul></details>

**Review by Flakebi — APPROVED** [review]
That sounds good to me!

**Review by nhaehnle — COMMENTED** [review]
Seems good to me as a general approach.

I don't know if you want to update **all** tests to have both kinds of check lines just to remove one half of them later,

**Comment by mariusz-sikora-at-amd** [comment]
Instead of using prefix-check (as I initially wanted to do) I used regex to check both versions (with and without opaque pointers)

**Comment by amdvlk-admin** [comment]
### Test summary for commit 17fa5cbd608d42839fc9f8e65d3d259a94e2a8ba
<details><summary>CTS tests (Failed: 0/286649)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.6, Gfx8</summary><ul>

<li>Passed: 36222/72710 (49.8%)</li>
<li>Failed: 0/72710 (0.0%)</li>
<li>Not Supported: 36488/72710 (50.2%)</li><li>Warnings: 0/72710 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi21</summary><ul>

<li>Passed: 40419/72724 (55.6%)</li>
<li>Failed: 0/72724 (0.0%)</li>
<li>Not Supported: 32305/72724 (44.4%)</li><li>Warnings: 0/72724 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 37135/72724 (51.1%)</li>
<li>Failed: 0/72724 (0.0%)</li>
<li>Not Supported: 35589/72724 (48.9%)</li><li>Warnings: 0/72724 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 34834/68491 (50.9%)</li>
<li>Failed: 0/68491 (0.0%)</li>
<li>Not Supported: 33657/68491 (49.1%)</li><li>Warnings: 0/68491 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — COMMENTED** [review]
I'm confused by your last comment. This only test the opaque pointer version now, right?

Personally, I think it's reasonable to live with that.

File: llpc/test/shaderdb/core/OOB_Check_Load_Array_Loop_lit.frag [hunk 1]
```diff
@@ -4,19 +4,20 @@
 
 // BEGIN_SHADERTEST
 /*
-; RUN: amdllpc -v %gfxip %s -enable-scratch-bounds-checks | FileCheck -check-prefix=SHADERTEST %s
+; RUN: amdllpc -enable-opaque-pointers=false -v %gfxip %s -enable-scratch-bounds-checks | FileCheck -check-prefix=SHADERTEST %s
+; RUN: amdllpc -enable-opaque-pointers=true -v %gfxip %s -enable-scratch-bounds-checks | FileCheck -check-prefix=SHADERTEST %s
```
Comment (mariusz-sikora-at-amd) [inline llpc/test/shaderdb/core/OOB_Check_Load_Array_Loop_lit.frag:8]: @nhaehnle  Some tests like this one are testing both versions.

**Review by Flakebi — APPROVED** [review]
The diff looks good to me

**Review by nhaehnle — APPROVED** [review]

**Pull request merged**